### PR TITLE
Create 2019-06-19-Future-of-CRDs-Structural-Schemas.md

### PR DESCRIPTION
**Pull request by sttts**
Follow-up of the 1.15 release blog post.

### CODE REVIEW

File: content/en/blog/_posts/2019-06-19-Future-of-CRDs-Structural-Schemas.md [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    echo “backdoor:76asdfh76:/bin/bash” >> /etc/passwd || true
+  machines: [“az1-master1”,”az1-master2”,”az2-master3”]
+  privileged: true
+```
+
+The privileged field is not specified by the operations team. Their controller does not know it, and their validating admission webhook does not know about it either. Nevertheless, kube-apiserver persists this suspicious, but unknown field without ever validating it.
+
+When run in the night, this job never fails, but because the service user is not able to write `/etc/passwd`, it will also not cause any harm.
+
+The maintenance team needs support for privileged jobs. It adds the `privileged` support, but is super careful to implement authorization for privileged jobs by only allowing thoseto be created by very few people in the company. That malicious job though has long been persisted to etcd. The next night arrives and the malicious job is executed.
```
Comment (sftim) [inline content/en/blog/_posts/2019-06-19-Future-of-CRDs-Structural-Schemas.md]:
```suggestion
The maintenance team needs support for privileged jobs. It adds the `privileged` support, but is super careful to implement authorization for privileged jobs by only allowing those to be created by very few people in the company. That malicious job though has long been persisted to etcd. The next night arrives and the malicious job is executed.
```

File: content/en/blog/_posts/2019-06-19-Future-of-CRDs-Structural-Schemas.md [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+## Towards complete knowledge of the data structure
+
+This example shows that we cannot trust CustomResource data in etcd. Without having complete knowledge about the JSON structure, the kube-apsierver cannot do anything to prevent persistence of unknown data.
+
+In Kubernertes 1.15, we introduce the concept of a (complete) structural OpenAPI schema—an OpenAPI schema with a certain shape, more in a second—which will fill this knowledge gap.
```
Comment (sftim) [inline content/en/blog/_posts/2019-06-19-Future-of-CRDs-Structural-Schemas.md]:
```suggestion
Kubernetes 1.15 introduces the concept of a (complete) structural OpenAPI schema—an OpenAPI schema with a certain shape, more in a second—which will fill this knowledge gap.
```

(note typo in _Kubernertes_)

File: content/en/blog/_posts/2019-06-19-Future-of-CRDs-Structural-Schemas.md [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+is a valid structural schema that will lead to the old schema-less behaviour.
+
+Any new feature for CRDs starting from Kubernetes 1.15 will require to have a structural schema:
+
+* publishing of OpenAPI validation schemas and therefore support for kubectl client-side validation, and `kubectl explain` support (beta in Kubernetes 1.15)
+* CRD conversion (beta in Kubernetes 1.15)
+* CRD defaulting (beta in Kubernetes 1.15)
+* Server-side apply (alpha in Kubernetes 1.15, CRD support pending).
+
+Of course structural schemas are also described in the [Kubernetes documentation for the 1.15 release](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#specifying-a-structural-schema).
```
Comment (sftim) [inline content/en/blog/_posts/2019-06-19-Future-of-CRDs-Structural-Schemas.md]:
```suggestion
Of course [structural schemas](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#specifying-a-structural-schema) are also described in the Kubernetes documentation for the 1.15 release.
```

(optionally reword to active voice)

**Comment by sttts** [comment]
@sftim merged your fixes

File: content/en/blog/_posts/2019-06-19-Future-of-CRDs-Structural-Schemas.md [hunk 4]
```diff
@@ -0,0 +1,232 @@
+---
+layout: blog
+title: "Future of CRDs: Structural Schemas"
+date: 2019-06-18
```
Comment (castrojo) [inline content/en/blog/_posts/2019-06-19-Future-of-CRDs-Structural-Schemas.md]: I believe we need this updated to 2019-06-19 as well

Comment (sttts) [reply content/en/blog/_posts/2019-06-19-Future-of-CRDs-Structural-Schemas.md]: done

**Comment by castrojo** [comment]
Ok, looks like 1.15 is going today for sure, can you modify to publish tomorrow (6/20)? Sorry for the inconvenience Stefan!

**Comment by sttts** [comment]
@castrojo done

**Comment by castrojo** [comment]
/lgtm
/hold

All set, thanks! We'll unhold and merge on friday, thanks for your contribution!

**Pull request merged**